resolvers/event.js

```diff
@@ -17,41 +36,230 @@ const eventResolvers = {
+        }]
+      });
+    },
+    addAmbulanceToEvent: async (parent, args) => {
+      // Checking if event exists
+      const event = await db.event.findByPk(args.eventId);
+      if (!event) {
+        throw new Error("Invalid event ID");
+      }
+
+      // Checking if ambulance exists
+      const ambulance = await db.ambulance.findByPk(args.ambulanceId['id']);
+      if (!ambulance) {
+        throw new Error("Invalid ambulance ID");
+      }
+      
+      // Checking if association between ambulance and event already exists
+      const alreadyExists = await db.eventAmbulances.count({
+        where: {
+          eventId: args.eventId,
+          ambulanceId: args.ambulanceId['id']
+        }
+      })
+
+      if (alreadyExists > 0) {
+        throw new Error("This ambulance is already assigned to this event")
+      }
+
+      // Creating association in eventAmbulances junction table
+      await db.eventAmbulances.create({
+        eventId: args.eventId,
+        ambulanceId: args.ambulanceId['id']
+      })
+
+      // Returning new event
+      return db.event.findByPk(args.eventId, { 
+        include: [{
+          model: db.ambulance,
+          attributes: ['id', 'vehicleNumber', 'createdAt', 'updatedAt'],
+        },
+        {
+          model: db.hospital,
+          attributes: ['id', 'name', 'createdAt', 'updatedAt']
+        }]
+      });
     },
-    deleteEvent: (parent, args) => {
+    addHospitalToEvent: async (parent, args) => {
+      // Checking if event exists
+      const event = await db.event.findByPk(args.eventId);
+      if (!event) {
+        throw new Error("Invalid event ID");
+      }
+      
+      // Checking if hospital exists
+      const hospital = await db.hospital.findByPk(args.hospitalId['id']);
+      if (!event) {
+        throw new Error("Invalid hospital ID");
+      }
+
+      // Checking if association between hospital and event already exists
+      const alreadyExists = await db.eventHospitals.count({
+        where: {
+          eventId: args.eventId,
+          hospitalId: args.hospitalId['id']
+        }
+      })
+      if (alreadyExists > 0) {
+        throw new Error("This hospital is already assigned to this event")
+      }
+
+      // Creating association in eventHospitals junction table
+      await db.eventHospitals.create({
+        eventId: args.eventId,
+        hospitalId: args.hospitalId['id']
+      })
+
+      // Returning new event
+      return db.event.findByPk(args.eventId, { 
+        include: [{
+          model: db.ambulance,
+          attributes: ['id', 'vehicleNumber', 'createdAt', 'updatedAt'],
+        },
+        {
+          model: db.hospital,
+          attributes: ['id', 'name', 'createdAt', 'updatedAt']
+        }]
+      });   
+    },
+    deleteEvent: async (parent, args) => {
       // Return status for destroy
       // 1 for successful deletion, 0 otherwise
+      await db.eventAmbulances.destroy({
```

Comment:
Wonder if we can set up delete cascades to deal foreign key deletions, but this is good for now!